### PR TITLE
Improve assert function

### PR DESCRIPTION
**Pull request by dborsatto**

### CODE REVIEW

File: services/contentful.js [hunk 1]
```diff
@@ -140,23 +140,31 @@ module.exports.getCoursesByCategory = assert((category, locale = 'en-US', api =
...
+ * This is done for easily catching 404 errors.
+ * @param  {string}   context The type of result the function is looking for
+ * @param  {Function} fn      The function to wrap
+ * @return {Object}           The result of `fn`, if not empty
+ * @throws {Error}    When `fn` returns an empty result
+ */
+function throwOnEmptyResult (context, fn) {
+  return function (...params) {
+    return fn(...params)
+      .then((data) => {
```
Comment (axe312ger) [inline services/contentful.js]: await here and turn the function in to an async one plz

Comment (dborsatto) [reply services/contentful.js]: I'd rather not tbh, I just changed the name of the function to make it more readable and replaced the parameter names (which were incorrect) with a simple `..params`. Changing the actual behavior of the function is beyond the scope of this PR.

Comment (axe312ger) [reply services/contentful.js]: U are right here, just realised we have a mix of styles in here. Mostly async/await but there are a few pure promises around. Out of scope ✅

**Comment by dborsatto** [comment]
@axe312ger CI is failing because the app's not been updated to reflect the latest tests for the modal, we can park this PR until that's fixed, then I'll rebase this branch so we don't merge a branch with failing CI 👍

**Comment by dborsatto** [comment]
@Khaledgarbaya can you take a quick look?

**Pull request merged**